src/deluge/model/clip/audio_clip.cpp

```diff
@@ -1123,26 +1123,40 @@ void AudioClip::setPos(ModelStackWithTimelineCounter* modelStack, int32_t newPos
 }
 
 bool AudioClip::shiftHorizontally(ModelStackWithTimelineCounter* modelStack, int amount) {
+	// No horizontal shift when recording
+	if (recorder) return false;
+	// No horizontal shift when no sample is loaded
+	if (!sampleHolder.audioFile) return false;
+
+	int64_t newStartPos = int64_t(sampleHolder.startPos) - getSamplesFromTicks(amount);
+	uint64_t sampleLength = ((Sample*)sampleHolder.audioFile)->lengthInSamples;
+
+	if (newStartPos < 0 || newStartPos > sampleLength) {
```

Comment:
`startPos == sampleLength` is OK?

Reply:
I had the same thought, but the sample marker editor allows it, so i assume it is ok
https://github.com/SynthstromAudible/DelugeFirmware/blob/9701434a2fd5c229f0a88ab8db43a9ed200092aa/src/deluge/gui/ui/sample_marker_editor.cpp#L220-L221